packages/server-core/src/user/magic-link/magic-link.class.ts

```diff
@@ -6,8 +6,8 @@ Version 1.0. (the "License"); you may not use this file except in compliance
 with the License. You may obtain a copy of the License at
```

Comment:
It seems like currently, it removing guest identity-provider when a magic link is send. This is not correct. The guest identity provider should be removed when a user login using url that he got in his email.

Please have a look of chat we had regarding making these changes in `login.class.ts`